samples, tests for llext build fails on mr_canhubk3

**Describe the bug**
`samples\subsys\llext`, `tests\subsys\llext` build fails on mr_canhubk3


**To Reproduce**
Steps to reproduce the behavior:
1. west build -p -b mr_canhubk3 -T tests\subsys\llext\llext.simple.arm
2. see error

**Expected behavior**
No build issue

**Impact**
Main development blocked when this test is triggered in CI for the target board

**Logs and console output**
```
In file included from D:/upstream/zephyr/include/zephyr/sys/util_macro.h:34,
                 from D:/upstream/zephyr/include/zephyr/sys/atomic.h:16,
                 from D:/upstream/zephyr/include/zephyr/kernel_structs.h:24,
                 from D:/upstream/zephyr/include/zephyr/arch/arm/arch_inlines.h:10,
                 from D:/upstream/zephyr/include/zephyr/arch/arch_inlines.h:18,
                 from D:/upstream/zephyr/include/zephyr/sys/arch_interface.h:1228,
                 from D:/upstream/zephyr/include/zephyr/arch/cpu.h:12,
                 from D:/upstream/zephyr/include/zephyr/irq.h:15,
                 from D:/upstream/zephyr/drivers/interrupt_controller/intc_eirq_nxp_s32.c:9:
D:/upstream/zephyr/drivers/interrupt_controller/intc_eirq_nxp_s32.c:192:47: error: 'IP_SIUL2_0_BASE' undeclared here (not in a function); did you mean 'IP_SIUL2_BASE'?
  192 |         (((DT_REG_ADDR(DT_INST_PARENT(n))) == IP_SIUL2_##i##_BASE) ? i : 0)
      |                                               ^~~~~~~~~
D:/upstream/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
D:/upstream/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit 559424285f635e7bb1c6878d8475907942723249

**Additional context**
Root cause: because some shim drivers use macros defined in `soc.h` , but that header file is not included in the drivers.
Only see these samples/tests build fail because MPU is disabled. When MPU is enabled, `soc.h` is indirectly included by some MPU header files

--- a/drivers/interrupt_controller/intc_eirq_nxp_s32.c
+++ b/drivers/interrupt_controller/intc_eirq_nxp_s32.c
@@ -6,6 +6,7 @@
 
 #define DT_DRV_COMPAT nxp_s32_siul2_eirq
 
+#include <soc.h>
 #include <zephyr/irq.h>
 #include <zephyr/sys/sys_io.h>
 #include <zephyr/drivers/pinctrl.h>

--- a/drivers/serial/uart_nxp_s32_linflexd.c
+++ b/drivers/serial/uart_nxp_s32_linflexd.c
@@ -6,6 +6,7 @@
 
 #define DT_DRV_COMPAT nxp_s32_linflexd
 
+#include <soc.h>
 #include <zephyr/irq.h>
 #include <zephyr/drivers/uart.h>
 #include <zephyr/drivers/pinctrl.h>

